Port gelu to structured

Freey0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58661 Port silu_backward to structured
* #58665 Port gelu_backward to structured
* **#58664 Port gelu to structured**
* #58663 Port hardshrink to structured
* #58662 Port hardshrink_backward and softshrink_backward to structured



Differential Revision: [D28572533](https://our.internmc.facebook.com/intern/diff/D28572533)

Freey0 committed:
Port gelu to structured

[ghstack-poisoned]

ezyang commented on aten/src/ATen/native/Activation.cpp:
```diff
@@ -153,6 +153,11 @@ TORCH_META_FUNC(softshrink_backward) (
 }
 
+TORCH_META_FUNC(gelu) (const Tensor & self) {
+  set_output(0, self.sizes(), {}, self.options().memory_format(LEGACY_CONTIGUOUS_MEMORY_FORMAT), {});
```
same deal as https://github.com/pytorch/pytorch/pull/58662/files#r636167323

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572533).

Freey0 committed:
Update on "Port gelu to structured"


Differential Revision: [D28572533](https://our.internmc.facebook.com/intern/diff/D28572533)

[ghstack-poisoned]

Freey0 committed:
Update on "Port gelu to structured"




Differential Revision: [D28572533](https://our.internmc.facebook.com/intern/diff/D28572533)

[ghstack-poisoned]

Freey0 committed:
Update on "Port gelu to structured"




Differential Revision: [D28572533](https://our.internmc.facebook.com/intern/diff/D28572533)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572533).

Freey0 committed:
Update on "Port gelu to structured"




Differential Revision: [D28572533](https://our.internmc.facebook.com/intern/diff/D28572533)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28572533).
